azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -211,7 +211,8 @@ private void deployArtifacts(IAppService target, WebAppConfig config) throws Azu
         // call correspond deploy method when deploy artifact only
         if (resources.size() == 1) {
             final Pair<File, DeployType> resource = resources.get(0);
-            target.deploy(resource.getRight(), resource.getLeft());
+            final DeployType deployType = target.getRuntime().getWebContainer() == WebContainer.JBOSS_72 ? DeployType.EAR : resource.getRight();
```

Comment:
why we need to override deployType?

Reply:
We will use file type as the deploy type by default, however, JBOSSEAP runtime both support `ear` and `war` artifact, but WarDeploy didn't support JBOSSEAP runtime(service will throw exception when target runtime is joss), so use EAR deploy for JBOSSEAP runtime